src/CDPVault.sol

```diff
@@ -1232,6 +1234,16 @@ abstract contract CDPVault is AccessControl, Pause, Permission, InterestRateMode
 
         // calculate the amount of collateral to exchange
         collateralToExchange = wdiv(creditToExchange, cache.settlementRate);
+
+        // bound the amount of collateral to exchange and account for accrued bad debt (if any)        
+        if (
+            collateralToExchange >= position.collateral && 
```

Comment:
@F2E4D1 @A6957E  Would this condition here be a problem if from precision loss `collateralToExchange` would be larger than `position.collateral` but `position.normalDebt` is less or equal to `deltaNormalDebt` then `modifyPosition` will revert due to collateral underflow.